NullPointerException IChromatogramSelection.getChromatogram() in ChromatogramReferencesUI initialize

This is an annoying follow-up error when the chromatogram is null.

```
!ENTRY org.eclipse.e4.ui.workbench 4 0 2026-02-06 08:41:47.727
!MESSAGE Unable to create class 'org.eclipse.chemclipse.ux.extension.xxd.ui.editors.ChromatogramEditorMSD' from bundle '649'
!STACK 0
org.eclipse.e4.core.di.InjectionException: java.lang.NullPointerException: Cannot invoke "org.eclipse.chemclipse.model.selection.IChromatogramSelection.getChromatogram()" because "chromatogramSelection" is null
	at org.eclipse.e4.core.internal.di.ConstructorRequestor.execute(ConstructorRequestor.java:53)
	at org.eclipse.e4.core.internal.di.InjectorImpl.internalMake(InjectorImpl.java:399)
	at org.eclipse.e4.core.internal.di.InjectorImpl.make(InjectorImpl.java:319)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.make(ContextInjectionFactory.java:203)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.createFromBundle(ReflectionContributionFactory.java:90)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.doCreate(ReflectionContributionFactory.java:59)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.create(ReflectionContributionFactory.java:42)
	at org.eclipse.ui.internal.WorkbenchContributionFactory.create(WorkbenchContributionFactory.java:46)
	at org.eclipse.e4.ui.workbench.renderers.swt.ContributedPartRenderer.createWidget(ContributedPartRenderer.java:133)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createWidget(PartRenderingEngine.java:987)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:658)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:762)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$2.run(PartRenderingEngine.java:726)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createGui(PartRenderingEngine.java:710)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.lambda$0(PartServiceImpl.java:105)
	at org.eclipse.e4.ui.services.internal.events.UIEventHandler.lambda$0(UIEventHandler.java:38)
	at org.eclipse.swt.widgets.Synchronizer.syncExec(Synchronizer.java:183)
	at org.eclipse.ui.internal.UISynchronizer.syncExec(UISynchronizer.java:136)
	at org.eclipse.swt.widgets.Display.syncExec(Display.java:5999)
	at org.eclipse.e4.ui.workbench.swt.DisplayUISynchronize.syncExec(DisplayUISynchronize.java:34)
	at org.eclipse.e4.ui.services.internal.events.UIEventHandler.handleEvent(UIEventHandler.java:38)
	at org.eclipse.equinox.internal.event.EventHandlerWrapper.handleEvent(EventHandlerWrapper.java:206)
	at org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent(EventHandlerTracker.java:201)
	at org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent(EventHandlerTracker.java:1)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:151)
	at org.eclipse.equinox.internal.event.EventAdminImpl.dispatchEvent(EventAdminImpl.java:132)
	at org.eclipse.equinox.internal.event.EventAdminImpl.sendEvent(EventAdminImpl.java:73)
	at org.eclipse.equinox.internal.event.EventComponent.sendEvent(EventComponent.java:48)
	at org.eclipse.e4.ui.services.internal.events.EventBroker.send(EventBroker.java:55)
	at org.eclipse.e4.ui.internal.workbench.UIEventPublisher.notifyChanged(UIEventPublisher.java:61)
	at org.eclipse.emf.common.notify.impl.BasicNotifierImpl.eNotify(BasicNotifierImpl.java:424)
	at org.eclipse.e4.ui.model.application.ui.impl.ElementContainerImpl.setSelectedElementGen(ElementContainerImpl.java:168)
	at org.eclipse.e4.ui.model.application.ui.impl.ElementContainerImpl.setSelectedElement(ElementContainerImpl.java:187)
	at org.eclipse.e4.ui.internal.workbench.ModelServiceImpl.showElementInWindow(ModelServiceImpl.java:650)
	at org.eclipse.e4.ui.internal.workbench.ModelServiceImpl.bringToTop(ModelServiceImpl.java:614)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.delegateBringToTop(PartServiceImpl.java:805)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.activate(PartServiceImpl.java:773)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.activate(PartServiceImpl.java:695)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.activate(PartServiceImpl.java:690)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.showPart(PartServiceImpl.java:1268)
	at org.eclipse.e4.ui.internal.workbench.ApplicationPartServiceImpl.showPart(ApplicationPartServiceImpl.java:200)
	at org.eclipse.chemclipse.ux.extension.ui.provider.ISupplierFileEditorSupport.openEditor(ISupplierFileEditorSupport.java:202)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.part.support.SupplierEditorSupport.openEditor(SupplierEditorSupport.java:167)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.part.support.SupplierEditorSupport.openEditor(SupplierEditorSupport.java:199)
	at org.eclipse.chemclipse.ux.extension.ui.provider.AbstractSupplierFileEditorSupport.openEditor(AbstractSupplierFileEditorSupport.java:38)
	at org.eclipse.chemclipse.ux.extension.ui.swt.MultiDataExplorerTreeUI.openEditorWithSupplier(MultiDataExplorerTreeUI.java:785)
	at org.eclipse.chemclipse.ux.extension.ui.swt.MultiDataExplorerTreeUI.openEditor(MultiDataExplorerTreeUI.java:770)
	at org.eclipse.chemclipse.ux.extension.ui.swt.MultiDataExplorerTreeUI.handleDoubleClick(MultiDataExplorerTreeUI.java:173)
	at org.eclipse.chemclipse.ux.extension.ui.swt.MultiDataExplorerTreeUI.lambda$2(MultiDataExplorerTreeUI.java:456)
	at org.eclipse.jface.viewers.StructuredViewer$1.run(StructuredViewer.java:781)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:169)
	at org.eclipse.jface.viewers.StructuredViewer.fireDoubleClick(StructuredViewer.java:778)
	at org.eclipse.jface.viewers.AbstractTreeViewer.handleDoubleSelect(AbstractTreeViewer.java:1578)
	at org.eclipse.jface.viewers.StructuredViewer$4.widgetDefaultSelected(StructuredViewer.java:1208)
	at org.eclipse.jface.util.OpenStrategy.fireDefaultSelectionEvent(OpenStrategy.java:271)
	at org.eclipse.jface.util.OpenStrategy$1.handleEvent(OpenStrategy.java:328)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:91)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:5894)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1656)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:5109)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4546)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1147)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1038)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:677)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:583)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:173)
	at org.eclipse.chemclipse.rcp.app.ui.internal.support.ApplicationSupportDefault.start(ApplicationSupportDefault.java:26)
	at org.eclipse.chemclipse.rcp.app.ui.Application.start(Application.java:28)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:219)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:149)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:115)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:467)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:298)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:615)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:563)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1415)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1387)
Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.chemclipse.model.selection.IChromatogramSelection.getChromatogram()" because "chromatogramSelection" is null
	at org.eclipse.chemclipse.ux.extension.xxd.ui.swt.ChromatogramReferencesUI.update(ChromatogramReferencesUI.java:122)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.swt.editors.ExtendedChromatogramUI.updateChromatogramSelection(ExtendedChromatogramUI.java:379)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.editors.AbstractChromatogramEditor.initialize(AbstractChromatogramEditor.java:271)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.editors.AbstractChromatogramEditor.<init>(AbstractChromatogramEditor.java:150)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.editors.ChromatogramEditor.<init>(ChromatogramEditor.java:29)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.editors.ChromatogramEditorMSD.<init>(ChromatogramEditorMSD.java:37)
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
	at org.eclipse.e4.core.internal.di.ConstructorRequestor.execute(ConstructorRequestor.java:41)
	... 84 more
```

Because the stack trace is so long, it hides the actual problem:

```
!ENTRY org.eclipse.chemclipse.ux.extension.xxd.ui 4 0 2026-02-06 08:41:45.974
!MESSAGE The chromatogram must not be null.
!STACK 0
org.eclipse.chemclipse.model.exceptions.ChromatogramIsNullException: The chromatogram must not be null.
	at org.eclipse.chemclipse.model.selection.AbstractChromatogramSelection.<init>(AbstractChromatogramSelection.java:56)
	at org.eclipse.chemclipse.msd.model.core.selection.ChromatogramSelectionMSD.<init>(ChromatogramSelectionMSD.java:59)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.internal.runnables.ChromatogramImportRunnable.run(ChromatogramImportRunnable.java:96)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:123)
```

--- a/chemclipse/plugins/org.eclipse.chemclipse.ux.extension.xxd.ui/src/org/eclipse/chemclipse/ux/extension/xxd/ui/swt/ChromatogramReferencesUI.java
+++ b/chemclipse/plugins/org.eclipse.chemclipse.ux.extension.xxd.ui/src/org/eclipse/chemclipse/ux/extension/xxd/ui/swt/ChromatogramReferencesUI.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2018, 2025 Lablicate GmbH.
+ * Copyright (c) 2018, 2026 Lablicate GmbH.
  *
  * This program and the accompanying materials are made
  * available under the terms of the Eclipse Public License 2.0
@@ -119,13 +119,15 @@ public class ChromatogramReferencesUI extends Composite {
 		selectionChangeListener = comboChromatograms;
 		comboViewerReferences.addSelectionChangedListener(selectionChangeListener);
 
-		IChromatogram chromatogram = chromatogramSelection.getChromatogram();
-		List<IChromatogramSelection> chromatogramSelections = new ArrayList<>();
-		chromatogramSelections.add(chromatogramSelection);
-		for(IChromatogram chromatogramReference : chromatogram.getReferencedChromatograms()) {
-			chromatogramSelections.add(createChromatogramSelection(chromatogramReference));
+		if(chromatogramSelection != null) {
+			IChromatogram chromatogram = chromatogramSelection.getChromatogram();
+			List<IChromatogramSelection> chromatogramSelections = new ArrayList<>();
+			chromatogramSelections.add(chromatogramSelection);
+			for(IChromatogram chromatogramReference : chromatogram.getReferencedChromatograms()) {
+				chromatogramSelections.add(createChromatogramSelection(chromatogramReference));
+			}
+			comboChromatograms.data = chromatogramSelections;
 		}
-		comboChromatograms.data = chromatogramSelections;
 
 		comboViewerReferences.setInput(comboChromatograms.data);
 	}

